lib/board.rb

```diff
@@ -9,50 +9,31 @@ def initialize
     }
   end
 
-  def cell_empty?(selected_cell)
-    return true if @cells[selected_cell.to_sym] == ' '
+  def cell_empty?(cell)
+    return true if @cells[cell] == ' '
     false
   end
 
-  def game_won?(piece)
-    return true if win?(board_rows, piece) || win?(board_columns, piece) || win?(board_diagonals, piece)
-  end
-
-  def render
-    puts "#{@cells[:one]}|#{@cells[:two]}|#{@cells[:three]}"
-    puts '-----'
-    puts "#{@cells[:four]}|#{@cells[:five]}|#{@cells[:six]}"
-    puts '-----'
-    puts "#{@cells[:seven]}|#{@cells[:eight]}|#{@cells[:nine]}"
+  def set_cell(cell, piece)
+    @cells[cell] = piece
   end
 
-  def win?(win_type, piece)
-    win_type.each_slice(3) do |type|
-      return true if type.all? {|cell| cell == piece }
+  def game_won?(piece)
+    return true if win?(piece)
+  end
+
+  def win?(cell)
+    if @cells[:one] != ' ' && @cells[:one] == @cells[:two] && @cells[:two] == @cells[:three] ||
```

Comment:
I'm noticing a pattern here.

```
if x != ' ' && x == y && y == z
```

I'd recommend making a helper method that you can call to perform this logic. I think the interface would look something like

```
foo(x, y,  z)
```

Your code would be easier to read through with less chance of having mistyped or copy-pasted the wrong variable name somewhere. Also, it might reveal other higher-level patterns in this logic that could possibly make for more helper methods 👍